DataFormats/TrackerRecHit2D/src/VectorHit.cc

```diff
@@ -0,0 +1,451 @@
+    //if(curvORphi == "curvature") std::cout << "curvature error: " << errorCurvature << std::endl;
+
+
+  } else {
+std::cout << " straight line!" << std::endl;
+    return std::make_pair(0.0,0.0);
+  }
+  
+  if( curvORphi == "curvature" ) return std::make_pair(curvature,errorCurvature);
+  else if( curvORphi == "phi"  ) return std::make_pair(phi,0.0);
+  else return std::make_pair(0.0,0.0);
+
+}
+
+float VectorHit::transverseMomentum(const MagneticField* magField){
+
+  GlobalPoint center(0.0, 0.0, 0.0);
+  float magnT = magField->inTesla(center).mag();
+  double rho = 1./curvatureORphi("curvature").first;
+  return (0.003*magnT*rho);
+
+}
+
+float VectorHit::momentum(const MagneticField* magField){
+  return transverseMomentum(magField)/(1.*sin(theta()));
+}
+
+float VectorHit::theta(){
+  return globalDirection().theta();
+}
+
+AlgebraicMatrix VectorHit::projectionMatrix() const {
+
+//  static bool isInitialized=false;
+  static AlgebraicMatrix the4DProjectionMatrix(4, 5, 0);
+/*
+  static AlgebraicMatrix the2DPhiProjMatrix(2, 5, 0);
+  static AlgebraicMatrix the2DZProjMatrix(2, 5, 0);
+
+  if (!isInitialized) {
+    the4DProjectionMatrix[0][1] = 1;
+    the4DProjectionMatrix[1][2] = 1;
+    the4DProjectionMatrix[2][3] = 1;
+    the4DProjectionMatrix[3][4] = 1;
+
+    the2DPhiProjMatrix[0][1] = 1;
+    the2DPhiProjMatrix[1][3] = 1;
+
+    the2DZProjMatrix[0][2] = 1;
+    the2DZProjMatrix[1][4] = 1;
+
+    isInitialized= true;
+  }
+
+  if (dimension()==4) {
+    return the4DProjectionMatrix;
+  } else if (theProjection==phi) {
+    return the2DPhiProjMatrix;
+  } else if (theProjection==Z) {
+    return the2DZProjMatrix;
+  } else {
+    return AlgebraicMatrix();
+  }
+*/ return the4DProjectionMatrix; //ERICA:QUESTO NON ESISTEVA!!
+}
+
+
+LocalError VectorHit::localPositionError() const {
+  return LocalError(theCovMatrix[2][2],theCovMatrix[2][3],theCovMatrix[3][3]);
+}
+
+
+LocalError VectorHit::localDirectionError() const {
+  return LocalError(theCovMatrix[0][0],theCovMatrix[0][1],theCovMatrix[1][1]);
+}
+
+
+
+AlgebraicSymMatrix VectorHit::parametersError() const {
+
+  //think about a more efficient method
+  AlgebraicSymMatrix result(4);
+  for(int i = 0; i < theDimension; i++){
+    for(int j = 0; j < theDimension; j++){
+      result[i][j] = theCovMatrix[i][j];
+    }
+  }
+  return result;
+}
+
+std::ostream& operator<<(std::ostream& os, const VectorHit& vh) {
+
+  os << " VectorHit create in the DetId#: " << vh.geographicalId() << "\n" <<
+        " Vectorhit local position      : " << vh.localPosition() << "\n" <<
+        " Vectorhit local direction     : " << vh.localDirection() << "\n" <<
+        " Vectorhit global direction    : " << vh.globalDirection() << "\n" <<
+        //" Vectorhit theta               : " << vh.theta() << "\n" <<
+        //" Cov: " << vh.parametersError() << "\n" <<
+        //" Dim: " << vh.dimension() << "\n" <<
+        //" chi2/ndof: " << vh.chi2() << "/" << vh.degreesOfFreedom() << "\n" <<
```

Comment:
@JanFSchulte In case you decide to remove `degreesOfFreedom()`..

Reply:
Thanks, removed the /ndof part